solr/core/src/java/org/apache/solr/update/UpdateLog.java

```diff
@@ -376,7 +376,19 @@ public void init(PluginInfo info) {
    * for an existing log whenever the core or update handler changes.
    */
   public void init(UpdateHandler uhandler, SolrCore core) {
-    dataDir = core.getUlogDir();
+    if (dataDir != null) {
+      // append the "dataDir" to the instance ie  <str name="dir">abc</str> ->
+      // <solr.data.home>/<core>/abc
+      dataDir =
+          core.getCoreDescriptor()
+              .getInstanceDir()
+              .resolve(dataDir + "/" + core.getName())
+              .toString();
```

Comment:
As it stands, I'm pretty sure that based on config `<str name="dir">abc</str>`, with code added here, files will be placed in:
```
solr_data/[core_name]/abc/[core_name]/tlog/tlog.00000000000000001
```
which is I don't think what we want. The reason this has been basically working in tests is likely because an absolute path has been specified for `dataDir` (which, confusingly, is `abc` in this example); but `CoreDescriptor.getInstanceDir()` will be `solr_data/[core_name]/`, so if we specify a relative path as "dir", we'll end up double-namespacing.

The weird thing is that the current upstream _lack_ of namespacing under the specified tlog dir seems to imply an assumption that (if "dir" were respected at all, which afaict it currently is _not_), the specified "dir" would be under the core instanceDir -- but in order to achieve that, you'd have to hardcode it per core, which is not workable.

I really think we need a test or two here. I _think_ I understand what's going on with this, but tbh I'm really not sure.